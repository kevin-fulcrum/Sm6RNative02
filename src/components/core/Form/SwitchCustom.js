import React, {Component} from 'react';
import {StyleSheet, View, Switch} from 'react-native';
import Label from './Label';

const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
  },
});

export default class SwitchCustom extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      enable: false,
    };
  }

  toggleSwitch = () => {
    this.props.onValueChange && this.props.onValueChange();
  };

  render() {
    const {
      children,
      value,
      onChange,
      disabled,
      trackColor,
      iosBackgroundColor,
      onValueChange,
      thumbColor,
      ...props
    } = this.props;
    return (
      <View>
        <Label {...this.props}>
          <Switch
            {...props}
            trackColor={trackColor}
            thumbColor={thumbColor}
            ios_backgroundColor={iosBackgroundColor}
            onValueChange={this.toggleSwitch}
            value={value}
          />
        </Label>
      </View>
    );
  }
}
