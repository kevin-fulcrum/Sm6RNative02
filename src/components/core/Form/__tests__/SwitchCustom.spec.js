import React from 'react';
import {View, Switch} from 'react-native';
import {shallow} from 'enzyme';
import SwitchCustom from '../SwitchCustom';

const toggleSwitch = jest.fn();

const initialProps = {
  value: true,
  onChange: jest.fn(),
  disabled: false,
  trackColor: '#212121',
  iosBackgroundColor: '#ffffff',
  onValueChange: toggleSwitch(),
  thumbColor: '#cccccc',
};

const wrapper = shallow(
  <SwitchCustom {...initialProps} onValueChange={toggleSwitch} />,
);
describe('SwitchCustom component', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should SwitchCustom element in component', () => {
    expect(wrapper.find(View).length).toBe(1);
    expect(wrapper.find(Switch).length).toBe(1);
  });
  it('should match props', () => {
    expect(wrapper.props.value).not.toBeNull();
    expect(wrapper.props.onChange).not.toBeNull();
    expect(wrapper.props.disabled).not.toBeNull();
    expect(wrapper.props.trackColor).not.toBeNull();
    expect(wrapper.props.iosBackgroundColor).not.toBeNull();

    expect(wrapper.props.onValueChange).not.toBeNull();
    expect(wrapper.props.thumbColor).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(wrapper.find(SwitchCustom)).toMatchSnapshot();
  });
});
