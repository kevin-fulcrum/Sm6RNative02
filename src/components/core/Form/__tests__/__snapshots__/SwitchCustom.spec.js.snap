// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchCustom component should match snapshot 1`] = `null`;
